apps/heureka/src/lib/slices/createGlobalsSlice.test.jsx

```diff
@@ -0,0 +1,275 @@
+        showIssueDetail: useGlobalsShowIssueDetail(),
+        views: {
+          Services: useGlobalsQueryOptions("Services"),
+          IssueMatches: useGlobalsQueryOptions("IssueMatches"),
+          Components: useGlobalsQueryOptions("Components"),
+        },
+      }),
+      { wrapper }
+    )
+
+    expect(result.current).toEqual({
+      embedded: false,
+      apiEndpoint: "https://api.com",
+      queryClientFnReady: false,
+      activeView: "Services",
+      showPanel: constants.PANEL_NONE,
+      showServiceDetail: null,
+      showIssueDetail: null,
+      views: {
+        Services: { first: 20 },
+        IssueMatches: { first: 20 },
+        Components: { first: 20 },
+      },
+    })
+  })
+
+  it("should update queryClientFnReady", () => {
+    const { result } = renderHook(
+      () => ({
+        actions: useGlobalsActions(),
+        queryClientFnReady: useGlobalsQueryClientFnReady(),
+      }),
+      { wrapper }
+    )
+
+    act(() => {
+      result.current.actions.setQueryClientFnReady(true)
+    })
+
+    expect(result.current.queryClientFnReady).toBe(true)
+  })
+
+  it("should update query options for a specific view", () => {
+    const newOptions = { first: 30, filter: "test" }
+    const { result } = renderHook(
+      () => ({
+        actions: useGlobalsActions(),
+        queryOptions: useGlobalsQueryOptions("Services"),
+      }),
+      { wrapper }
+    )
+
+    act(() => {
+      result.current.actions.setQueryOptions("Services", newOptions)
+    })
+
+    expect(result.current.queryOptions).toEqual(newOptions)
+  })
+
+  it("should update embedded state", () => {
+    const { result } = renderHook(
+      () => ({
+        actions: useGlobalsActions(),
+        embedded: useGlobalsEmbedded(),
+      }),
+      { wrapper }
+    )
+
+    act(() => {
+      result.current.actions.setEmbedded(true)
+    })
+
+    expect(result.current.embedded).toBe(true)
+  })
+
+  it("should update apiEndpoint", () => {
+    const newEndpoint = "https://newapi.com"
+    const { result } = renderHook(
+      () => ({
+        actions: useGlobalsActions(),
+        apiEndpoint: useGlobalsApiEndpoint(),
+      }),
+      { wrapper }
+    )
+
+    act(() => {
+      result.current.actions.setApiEndpoint(newEndpoint)
+    })
+
+    expect(result.current.apiEndpoint).toBe(newEndpoint)
+  })
+
+  it("should update activeView", () => {
+    const { result } = renderHook(
+      () => ({
+        actions: useGlobalsActions(),
+        activeView: useGlobalsActiveView(),
+      }),
+      { wrapper }
+    )
+
+    act(() => {
+      result.current.actions.setActiveView("IssueMatches")
+    })
+
+    expect(result.current.activeView).toBe("IssueMatches")
+  })
+
+  it("should update showPanel", () => {
+    const { result } = renderHook(
+      () => ({
+        actions: useGlobalsActions(),
+        showPanel: useGlobalsShowPanel(),
+      }),
+      { wrapper }
+    )
+
+    act(() => {
```

Comment:
I am not sure that we need an act here ...

Reply:
I mean everywhere where you use it in a whole test file

Reply:
The same here, I have to use act, otherwise when I use the store actions directly, I get the following error, because here I render StoreProvider to instatiate the store context. 
` Warning: An update to TestComponent inside a test was not wrapped in act(...).
│ 
│ When testing, code that causes React state updates should be wrapped into act(...):
│ 
│ act(() => {
│   /* fire events that update state */
│ });
│ /* assert on the output */
│ 
│ This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
│     at TestComponent (/workspace/juno/node_modules/@testing-library/react/dist/pure.js:307:5)
│     at StoreProvider (/workspace/juno/apps/heureka/src/components/StoreProvider.jsx:19:26)
│     at wrapper (/workspace/juno/apps/heureka/src/lib/slices/createGlobalsSlice.test.jsx:14:22)`